scripts/check-changelog.py

```diff
@@ -41,11 +42,21 @@
         for d in (now, now + hour, now - hour)
     ))
 
-    for line in tools.changelog().split('\n'):
-        if line.strip() in acceptable_lines:
+    pattern = r'\n\d+\.\d+\.\d+ - \d\d\d\d-\d\d-\d\d\n'
+    all_versions = list(map(str.strip, re.findall(pattern, tools.changelog())))
```

Comment:
I like the date pattern better, but otherwise I find it easier to think about `re.findall` returning a list of strings.